samples/block_requests/plugin.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "proxy_wasm_intrinsics.h"
+#include <regex>
+
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    const auto& referer = getRequestHeader("Referer");
+
+    if (referer && referer->view() == "https://www.example.com/") {
+      sendLocalResponse(404, "", "Error - Not Found.\n", {});
+      return FilterHeadersStatus::StopAllIterationAndWatermark;
```

Comment:
@mpwarres Can you confirm that this is the desired return value for a plugin that makes use of sendLocalResponse?

Reply:
This is what is used in here in a similar use case: https://github.com/GoogleCloudPlatform/service-extensions-samples/blob/aa8eac4d3ea1685de70ab8d6bb92e2b2dc5ae138/samples/config_denylist/plugin.cc#LL64C35-L64C63